Library/Homebrew/language/python.rb

```diff
@@ -139,11 +139,45 @@ def virtualenv_create(venv_root, python = "python", formula = self)
         venv
       end
 
+      def needs_python2?
+        requirements.any? do |req|
+          req.is_a?(PythonRequirement) && !req.is_a?(Python3Requirement) &&
+            (req.required? || build.with?(req))
+        end ||
+          deps.any? do |dep|
+            dep.name == "python" && (dep.required? || build.with?(dep))
+          end
+      end
```

Comment:
Yeah, I agree. I think the main question is whether there are more ambiguous cases that should force an error unless it's specified manually. It seems reasonable to me that in the above scenario
```
depends_on :python if MacOS.version <= :snow_leopard
depends_on :python3
```
that the default would be python3 without forcing it to be specified manually. Ditto in the case of "without-python" and :python3. But I could see someone reasonably wanting those cases to raise unless manually specified.